### PR TITLE
Fixes #5907: gitlab_runner is not idempotent on first run after runner creation

### LINKED ISSUE #5907: gitlab_runner is not idempotent on first run after runner creation
### Summary

On the first run after a GitLab runner creation, Ansible reports a change, although the configuration has not changed in the meanwhile. All subsequent runs on the other hand are idempotent.

### Issue Type

Bug Report

### Component Name

gitlab_runner

### Ansible Version

```console (paste below)
ansible [core 2.14.1]
  config file = None
  configured module search path = ['/home/ansible/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ansible/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/ansible/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/ansible/.local/bin/ansible
  python version = 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
```

### Community.general Version

```console (paste below)
Collection        Version
----------------- -------
community.general 6.3.0
```

### Configuration

```console (paste below)
CONFIG_FILE() = None
```

### OS / Environment

Ubuntu 22.04

### Steps to Reproduce

Just create a GitLab runner and re-execute the playbook.

```yaml (paste below)
- name: Test playbook
  hosts: localhost
  become: false
  gather_facts: false
  tasks:
    - name: Ensure GitLab runner
      community.general.gitlab_runner:
        api_url: https://git.mycompany.com/
        api_token: "{{ gitlab_runner_api_token }}"
        registration_token: "{{ gitlab_runner_registration_token }}"
        name: Test GitLab Runner
        state: present
        project: loc/test
```

### Expected Results

Ansible should report no change on subsequent runs when the runner configuration has not changed.

### Actual Results

```console (paste below)
$ ansible-playbook test-gitlab-runner.yml -vvvv
ansible-playbook [core 2.14.1]
  config file = /home/ansible/automation/projects/provisioning/ansible.cfg
  configured module search path = ['/home/ansible/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ansible/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/ansible/automation/projects/provisioning/collections
  executable location = /home/ansible/.local/bin/ansible-playbook
  python version = 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
Using /home/ansible/automation/projects/provisioning/ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
script declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
auto declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
yaml declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
ini declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
toml declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
Loading collection community.general from /home/ansible/automation/projects/provisioning/collections/ansible_collections/community/general
Loading callback plugin default of type stdout, v2.0 from /home/ansible/.local/lib/python3.10/site-packages/ansible/plugins/callback/default.py
Loading collection ansible.posix from /home/ansible/.local/lib/python3.10/site-packages/ansible_collections/ansible/posix
Skipping callback 'default', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.
Loading callback plugin ansible.posix.profile_tasks of type aggregate, v2.0 from /home/ansible/.local/lib/python3.10/site-packages/ansible_collections/ansible/posix/plugins/callback/profile_tasks.py

PLAYBOOK: test-gitlab-runner.yml ***************************************************************************************************************************************************
Positional arguments: test-gitlab-runner.yml
verbosity: 4
connection: smart
timeout: 60
become_method: sudo
tags: ('all',)
inventory: ('/etc/ansible/hosts',)
forks: 5
1 plays in test-gitlab-runner.yml

PLAY [Test playbook] ***************************************************************************************************************************************************************

TASK [Ensure GitLab runner] ********************************************************************************************************************************************************
task path: /home/ansible/automation/projects/provisioning/test-gitlab-runner.yml:7
Saturday 28 January 2023  12:34:22 +0100 (0:00:00.053)       0:00:00.053 ******
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: ansible
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /tmp/.ansible-$USER/tmp `"&& mkdir "` echo /tmp/.ansible-$USER/tmp/ansible-tmp-1674905662.979175-3040807-195011873830039 `" && echo ansible-tmp-1674905662.979175-3040807-195011873830039="` echo /tmp/.ansible-$USER/tmp/ansible-tmp-1674905662.979175-3040807-195011873830039 `" ) && sleep 0'
Using module file /home/ansible/automation/projects/provisioning/collections/ansible_collections/community/general/plugins/modules/gitlab_runner.py
<127.0.0.1> PUT /home/ansible/.ansible/tmp/ansible-local-3040798bs03wu04/tmpq8stvhl6 TO /tmp/.ansible-ansible/tmp/ansible-tmp-1674905662.979175-3040807-195011873830039/AnsiballZ_gitlab_runner.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /tmp/.ansible-ansible/tmp/ansible-tmp-1674905662.979175-3040807-195011873830039/ /tmp/.ansible-ansible/tmp/ansible-tmp-1674905662.979175-3040807-195011873830039/AnsiballZ_gitlab_runner.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python3 /tmp/.ansible-ansible/tmp/ansible-tmp-1674905662.979175-3040807-195011873830039/AnsiballZ_gitlab_runner.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /tmp/.ansible-ansible/tmp/ansible-tmp-1674905662.979175-3040807-195011873830039/ > /dev/null 2>&1 && sleep 0'
changed: [localhost] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "access_level": "ref_protected",
            "active": true,
            "api_job_token": null,
            "api_oauth_token": null,
            "api_password": null,
            "api_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "api_url": "https://git.mycompany.com/",
            "api_username": null,
            "description": "Test GitLab Runner",
            "locked": false,
            "maximum_timeout": 3600,
            "name": "Test GitLab Runner",
            "owned": false,
            "project": "loc/test",
            "registration_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "run_untagged": true,
            "state": "present",
            "tag_list": [],
            "validate_certs": true
        }
    },
    "msg": "Successfully created or updated the runner Test GitLab Runner",
    "runner": {
        "id": 1232,
        "token": "7Qr-JaAzYS2LH9eAi2HV",
        "token_expires_at": null
    }
}

PLAY RECAP *************************************************************************************************************************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

Saturday 28 January 2023  12:34:23 +0100 (0:00:01.004)       0:00:01.057 ******
===============================================================================
Ensure GitLab runner -------------------------------------------------------------------------------------------------------------------------------------------------------- 1.00s
/home/ansible/automation/projects/provisioning/test-gitlab-runner.yml:7 -----------------------------------------------------------------------------------------------------------

$ ansible-playbook test-gitlab-runner.yml -vvvv
ansible-playbook [core 2.14.1]
  config file = /home/ansible/automation/projects/provisioning/ansible.cfg
  configured module search path = ['/home/ansible/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ansible/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/ansible/automation/projects/provisioning/collections
  executable location = /home/ansible/.local/bin/ansible-playbook
  python version = 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
Using /home/ansible/automation/projects/provisioning/ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
script declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
auto declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
yaml declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
ini declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
toml declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
Loading collection community.general from /home/ansible/automation/projects/provisioning/collections/ansible_collections/community/general
Loading callback plugin default of type stdout, v2.0 from /home/ansible/.local/lib/python3.10/site-packages/ansible/plugins/callback/default.py
Loading collection ansible.posix from /home/ansible/.local/lib/python3.10/site-packages/ansible_collections/ansible/posix
Skipping callback 'default', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.
Loading callback plugin ansible.posix.profile_tasks of type aggregate, v2.0 from /home/ansible/.local/lib/python3.10/site-packages/ansible_collections/ansible/posix/plugins/callback/profile_tasks.py

PLAYBOOK: test-gitlab-runner.yml ***************************************************************************************************************************************************
Positional arguments: test-gitlab-runner.yml
verbosity: 4
connection: smart
timeout: 60
become_method: sudo
tags: ('all',)
inventory: ('/etc/ansible/hosts',)
forks: 5
1 plays in test-gitlab-runner.yml

PLAY [Test playbook] ***************************************************************************************************************************************************************

TASK [Ensure GitLab runner] ********************************************************************************************************************************************************
task path: /home/ansible/automation/projects/provisioning/test-gitlab-runner.yml:7
Saturday 28 January 2023  12:34:26 +0100 (0:00:00.051)       0:00:00.051 ******
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: ansible
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /tmp/.ansible-$USER/tmp `"&& mkdir "` echo /tmp/.ansible-$USER/tmp/ansible-tmp-1674905666.289351-3040847-56819068716431 `" && echo ansible-tmp-1674905666.289351-3040847-56819068716431="` echo /tmp/.ansible-$USER/tmp/ansible-tmp-1674905666.289351-3040847-56819068716431 `" ) && sleep 0'
Using module file /home/ansible/automation/projects/provisioning/collections/ansible_collections/community/general/plugins/modules/gitlab_runner.py
<127.0.0.1> PUT /home/ansible/.ansible/tmp/ansible-local-30408433yz6d20y/tmplkxru1sf TO /tmp/.ansible-ansible/tmp/ansible-tmp-1674905666.289351-3040847-56819068716431/AnsiballZ_gitlab_runner.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /tmp/.ansible-ansible/tmp/ansible-tmp-1674905666.289351-3040847-56819068716431/ /tmp/.ansible-ansible/tmp/ansible-tmp-1674905666.289351-3040847-56819068716431/AnsiballZ_gitlab_runner.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python3 /tmp/.ansible-ansible/tmp/ansible-tmp-1674905666.289351-3040847-56819068716431/AnsiballZ_gitlab_runner.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /tmp/.ansible-ansible/tmp/ansible-tmp-1674905666.289351-3040847-56819068716431/ > /dev/null 2>&1 && sleep 0'
changed: [localhost] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "access_level": "ref_protected",
            "active": true,
            "api_job_token": null,
            "api_oauth_token": null,
            "api_password": null,
            "api_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "api_url": "https://git.mycompany.com/",
            "api_username": null,
            "description": "Test GitLab Runner",
            "locked": false,
            "maximum_timeout": 3600,
            "name": "Test GitLab Runner",
            "owned": false,
            "project": "loc/test",
            "registration_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "run_untagged": true,
            "state": "present",
            "tag_list": [],
            "validate_certs": true
        }
    },
    "msg": "Successfully created or updated the runner Test GitLab Runner",
    "runner": {
        "access_level": "ref_protected",
        "active": true,
        "architecture": null,
        "contacted_at": null,
        "description": "Test GitLab Runner",
        "groups": [],
        "id": 1232,
        "ip_address": "10.2.4.14",
        "is_shared": false,
        "locked": false,
        "maximum_timeout": 3600,
        "name": null,
        "online": null,
        "paused": false,
        "platform": null,
        "projects": [
            {
                "avatar_url": null,
                "created_at": "2023-01-27T12:32:44.132+01:00",
                "default_branch": "main",
                "description": null,
                "forks_count": 0,
                "http_url_to_repo": "https://git.mycompany.com/loc/test.git",
                "id": 1910,
                "last_activity_at": "2023-01-27T13:54:22.868+01:00",
                "name": "test",
                "name_with_namespace": "loc / test",
                "namespace": {
                    "avatar_url": null,
                    "full_path": "loc",
                    "id": 539,
                    "kind": "group",
                    "name": "loc",
                    "parent_id": null,
                    "path": "loc",
                    "web_url": "https://git.mycompany.com/groups/loc"
                },
                "path": "test",
                "path_with_namespace": "loc/test",
                "readme_url": "https://git.mycompany.com/loc/test/-/blob/main/README.md",
                "ssh_url_to_repo": "git@git.mycompany.com:loc/test.git",
                "star_count": 0,
                "tag_list": [],
                "topics": [],
                "web_url": "https://git.mycompany.com/loc/test"
            }
        ],
        "revision": null,
        "run_untagged": true,
        "runner_type": "project_type",
        "status": "never_contacted",
        "tag_list": [],
        "version": null
    }
}

PLAY RECAP *************************************************************************************************************************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

Saturday 28 January 2023  12:34:27 +0100 (0:00:01.050)       0:00:01.102 ******
===============================================================================
Ensure GitLab runner -------------------------------------------------------------------------------------------------------------------------------------------------------- 1.05s
/home/ansible/automation/projects/provisioning/test-gitlab-runner.yml:7 -----------------------------------------------------------------------------------------------------------

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by cfiehe**
##### SUMMARY
This fix introduces the new boolean option ``access_level_on_creation``. It controls, whether the value of ``access_level`` is used for runner registration or not. The option ``access_level`` has been ignored on registration so far and was only used on updates. The user is informed by a deprecation warning, if the option is unspecified. For reasons of compatibility ``false`` is assumed in that case. The option ``access_level_on_creation`` will switch to ``true`` in community.general 7.0.0.

Signed-off-by: Christoph Fiehe <c.fiehe@eurodata.de>

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
gitlab_runner

### CODE REVIEW

File: plugins/modules/gitlab_runner.py [hunk 1]
```diff
@@ -216,6 +216,7 @@ def create_or_update_runner(self, description, options):
                 'locked': options['locked'],
                 'run_untagged': options['run_untagged'],
                 'maximum_timeout': options['maximum_timeout'],
+                'access_level': options['access_level'],
```
Comment (nejch) [inline plugins/modules/gitlab_runner.py:219]: This looks ok to me, but I think it has a slight change in runner registration behavior: this module defines `ref_protected` as the default for creation, which I think is not the real default in GitLab's API if the attribute is not provided*.
https://github.com/ansible-collections/community.general/blob/13b9d949fc20378c42d8c5dcbfe3034c1894ddeb/plugins/modules/gitlab_runner.py#L331

But I guess this happens anyway after the first run with the current approach, since the update method will then change it, and that is the reason you opened the issue in the first place I assume. 

Just wanted to point this out but will let @felixfontein decide on the best approach.

*See https://docs.gitlab.com/ee/api/runners.html#register-a-new-runner

Comment (nejch) [reply plugins/modules/gitlab_runner.py:219]: A little side note: Not sure about the history behind this, but I think selecting `ref_protected` as the default here was a bit of an exotic choice. Of course that's more secure as it limits the runner to being available on protected branches only, but I'm not sure that's the more common use case for GitLab runners.

Comment (cfiehe) [reply plugins/modules/gitlab_runner.py:219]: That is good remark. I was not aware of that fact. From that point of view, it would be the best to also change the default value of `access_level` to `not_protected`. Because the `access_level` was not passed to the runner creation call before, people actually got an unprotected runner. That is not what you would expect from the documentation. But it is inline with the default behavior of the GitLab API.

Comment (nejch) [reply plugins/modules/gitlab_runner.py:219]: @cfiehe I agree that makes more sense and I see you've already incorporated it. The only thing now is that this is technically a breaking change so I'm eager to hear feedback from the collection maintainers (I'm only involved in the GitLab module here :))

Comment (felixfontein) [reply plugins/modules/gitlab_runner.py:219]: Changing the default value is a breaking change and cannot be merged (see also #582). Changing the default requires a longer deprecation period.

Generally I would avoid having a default *at all* though, then the default on creation would be "use whatever GitLab thinks is the default", and the default for updating would be "don't update that value". Whether you prefer to force a specific value for this option if none is provided, or whether not specifying a value means "I don't care / don't change", that's something for the module maintainers to decide. But it's nothing this bugfix PR should touch/change.

File: changelogs/fragments/5907-fix-gitlab_runner-not-idempotent.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+ - gitlab_runner - add missing parameter 'access_level' on runner creation and change default value to 'not_protected' (https://github.com/ansible-collections/community.general/issues/5907).
```
Comment (felixfontein) [inline changelogs/fragments/5907-fix-gitlab_runner-not-idempotent.yml:2]: I would make this a bit more explicit what this means for users:
```suggestion
 - gitlab_runner - add missing parameter ``access_level`` on runner creation. It has been ignored on creation so far and was only used on updating (https://github.com/ansible-collections/community.general/issues/5907).
```
(I cut out the default change since that shouldn't be part of this PR anyway.)

File: plugins/modules/gitlab_runner.py (hunk 1 above)
Comment (nejch) [reply plugins/modules/gitlab_runner.py:219]: @felixfontein the issue is that the bugfix, without removing the default, would also be a breaking change.

So maybe removing the default here is the easiest transition, but then the code will need to ensure that it's not sending null values to the API if no attribute is present, as GitLab sometimes doesn't like that.

Comment (felixfontein) [reply plugins/modules/gitlab_runner.py:219]: That's true, but at least it's restoring the documented behavior, and not changing the documented behavior *and* the implementation.

But yeah, it's also not helping that the current behavior has been there for quite a few years (it has always been like this since it was migrated to this repository). That's also (in part) why I didn't add the `backport-5` label.

So maybe the best way forward is:
1. Add a new option `access_level_on_creation` of type boolean with no explicit default, but implicit default `false`.
2. If that option is not specified, a deprecation warning will be triggered that this option will switch to `true` for the next major release (community.general 7.0.0).
3. Only set the access level on creation when `access_level_on_creation=true` is passed to the module.

It would be a rather short deprecation period, but at least users would know that something is changing, and (most importantly) can get the module to use the new behavior already now. What do you think about this approach?

(Deprecating the default value of `access_level` is something that should probably happen in a separate PR, as it's strictly not related to this problem.)

File: changelogs/fragments/5907-fix-gitlab_runner-not-idempotent.yml (hunk 2 above)
Comment (cfiehe) [reply changelogs/fragments/5907-fix-gitlab_runner-not-idempotent.yml]: Ok. sorry. I have not seen your comment. I have removed the default value of the parameter `access_level` on the last change, knowing that this is a breaking change. What you already stated before, maybe this kind of modification is out out scope of a PR.

I will wait with any further modifications until a decision was made, how to handle the problem with the missing parameter `access_level`.

Comment (cfiehe) [reply changelogs/fragments/5907-fix-gitlab_runner-not-idempotent.yml]: In order to revert this change in behavior, it "just" has to be decided which default value to use for `access_level`. The problem here is, that whatever the decision is, we always have a change in behavior. People you used that module only for initial runner creation, got an unprotected runner, whereas people who executed this module more than once got a protected runner. It is a dilemma.

Comment (felixfontein) [reply changelogs/fragments/5907-fix-gitlab_runner-not-idempotent.yml]: See https://github.com/ansible-collections/community.general/pull/5908#discussion_r1089756143, this works around that dilemma (by default there is no change in behavior, but you get a warning and can easily switch to the new behavior already).

File: changelogs/fragments/5907-fix-gitlab_runner-not-idempotent.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - gitlab_runner - add new boolean option ``access_level_on_creation``. It controls, whether the value of ``access_level`` is used for runner registration or not. The option ``access_level`` has been ignored on registration so far and was only used on updates (https://github.com/ansible-collections/community.general/issues/5907).
```
Comment (felixfontein) [inline changelogs/fragments/5907-fix-gitlab_runner-not-idempotent.yml:2]:
```suggestion
  - gitlab_runner - add new boolean option ``access_level_on_creation``. It controls, whether the value of ``access_level`` is used for runner registration or not. The option ``access_level`` has been ignored on registration so far and was only used on updates (https://github.com/ansible-collections/community.general/issues/5907, https://github.com/ansible-collections/community.general/pull/5908).
deprecated_features:
  - gitlab_runner - the default of the new option ``access_level_on_creation`` will change from ``false`` to ``true`` in community.general 7.0.0. This will cause ``access_level`` to be used during runner registration as well, and not only during updates (https://github.com/ansible-collections/community.general/pull/5908).
```

**Comment by cfiehe** [comment]
Thanks for the review. I have added the requested changes and rebased the branch.

File: plugins/modules/gitlab_runner.py [hunk 4]
```diff
@@ -84,12 +84,22 @@
...
       - If set to C(not_protected), runner can pick up jobs from both protected and unprotected branches.
     required: false
     default: ref_protected
     choices: ["ref_protected", "not_protected"]
     type: str
+  access_level_on_creation:
+    description:
+      - Whether the runner should be registered with an access level or not.
+      - If set to C(true), the value of I(access_level) is used for runner registration.
+      - If set to C(false), GitLab registers the runner with the default access level.
```
Comment (felixfontein) [inline plugins/modules/gitlab_runner.py:99]:
```suggestion
      - If set to C(false), GitLab registers the runner with the default access level.
      - The current default of this option is C(false). This default is deprecated and will change to C(true) in commuinty.general 7.0.0.
```

**Review by felixfontein — COMMENTED** [review]
Besides this looks good to me! @nejch what do you think?

File: plugins/modules/gitlab_runner.py (hunk 4 above)
Comment (cfiehe) [reply plugins/modules/gitlab_runner.py:99]: Added and pushed. Thanks.

**Review by nejch — APPROVED** [review]
> Besides this looks good to me! @nejch what do you think?

@felixfontein @cfiehe LGTM like this! It might make sense to track the `access_level` default in a follow-up to align with GitLab eventually, or @cfiehe if you're willing to open another PR for that :)

**Comment by felixfontein** [comment]
@cfiehe thanks a lot for your contribution!
@nejch thanks a lot for reviewing this!

Creating a follow-up issue or PR sounds good to me. The first step will be removing the current default and documenting it manually, and using a similar mechanism as for the new option in this PR (if the default is not set, call `module.deprecate(...)` and explicitly set it to the old default). The deprecation period for this default needs to be longer though, it should be for c.g 8.0.0.
